test/registry/finalizeExit.js

```diff
@@ -0,0 +1,245 @@
+      assert.strictEqual(listingStruct[5].toString(), '0', 'exit time was initialized');
+    });
+
+    it('should not allow a listing to finalize exit when time is not up', async () => {
+      const listing = utils.getListingHash('hangouts.com');
+
+      const initialApplicantTokenHoldings = await token.balanceOf.call(applicant);
+
+      await utils.addToWhitelist(listing, paramConfig.minDeposit, applicant, registry);
+
+      const isWhitelisted = await registry.isWhitelisted.call(listing);
+      assert.strictEqual(isWhitelisted, true, 'the listing was not added to the registry');
+      await registry.initExit(listing, { from: applicant });
+      // blockTimestamp is used to calculate when the applicant's exit time is up
+      const blockTimestamp = new BigNumber(await utils.getBlockTimestamp());
+      try {
+        await registry.finalizeExit(listing, { from: applicant });
+        assert(false, 'exit succeeded when it should have failed due to time not being up');
+      } catch (err) {
+        const errMsg = err.toString();
+        assert(utils.isEVMException(err), errMsg);
+      }
+      const isWhitelistedAfterExit = await registry.isWhitelisted.call(listing);
+      assert(isWhitelistedAfterExit, true, 'the listing was removed on finalizeExit');
+
+      const finalApplicantTokenHoldings = await token.balanceOf.call(applicant);
+      assert.strictEqual(
+        finalApplicantTokenHoldings.toString(),
+        initialApplicantTokenHoldings.sub(paramConfig.minDeposit).toString(),
+        'the applicant\'s tokens were returned in spite of failing to exit',
+      );
+      const listingStruct = await registry.listings.call(listing);
+      assert.strictEqual(listingStruct[5].toString(), blockTimestamp.add(paramConfig.exitTimeDelay).toString(), 'exit time was not initialized');
+    });
+
+    it('should not allow a listing to finalize an exit when a challenge does exist', async () => {
+      const listing = utils.getListingHash('520.com');
+
+      const initialApplicantTokenHoldings = await token.balanceOf.call(applicant);
+
+      await utils.addToWhitelist(listing, paramConfig.minDeposit, applicant, registry);
+
+      const isWhitelisted = await registry.isWhitelisted.call(listing);
+      assert.strictEqual(isWhitelisted, true, 'the listing was not added to the registry');
+
+      await registry.initExit(listing, { from: applicant });
+      // blockTimestamp is used to calculate when the applicant's exit time is up
+      const blockTimestamp = new BigNumber(await utils.getBlockTimestamp());
+      await utils.increaseTime(paramConfig.exitTimeDelay + 1);
+      await registry.challenge(listing, '', { from: challenger });
+      // Make an assertion to prove that challenge does exit
+      const initialListingStruct = await registry.listings.call(listing);
+      assert.notStrictEqual(initialListingStruct[4].toString(), '0', 'Challenge was never created');
```

Comment:
I don't think you need to test that `challenge` works properly, as there are already tests for that function.